src/datagrid/DataGrid.tsx

```diff
@@ -27,40 +29,46 @@ import {classNames, allTrueOnKey} from "../utils";
  * @param {selectionType} row selection type that is multi or single
  * @param {pagination} pagination support
  * @param {columns} column details
- * @param {data} rows data
+ * @param {rows} rows data
  * @param {footer} footer component
  * @param {onRowSelect} Function which will gets called on select/deselect of rows
  * @param {onSelectAll} Function which will gets called on select/deselect of all rows
+ * @param {rowType} Expandable or compact row type
+ * @param {itemText} label to display for all items
  */
 type DataGridProps = {
     className?: string;
     style?: any;
     selectionType?: GridSelectionType;
     pagination?: boolean;
     columns: DataGridColumn[];
-    data: DataGridRow[];
+    rows?: DataGridRow[];
     footer?: DataGridFooter;
     onRowSelect?: Function;
     onSelectAll?: Function;
+    keyfield?: string;
+    rowType?: GridRowType;
+    itemText?: string;
 };
 
 /**
  * type for DataGridColumn :
- * @param {content} column data
+ * @param {columnName} column data
  * @param {sort} does colum support sorting
- * @param {filter} does colum support filtering
  * @param {className} CSS class name
  * @param {columns} column details
  * @param {style} CSS style
- * @param {onFilter} Function for custom filtering
+ * @param {width} width of column
+ * @param {filter} Filter component
  */
-type DataGridColumn = {
-    content: any;
-    sort?: boolean;
-    filter?: boolean;
+export type DataGridColumn = {
+    columnName: string;
+    columnID?: number; // For internal use
+    sort?: DataGridSort;
     className?: string;
     style?: any;
-    onFilter?: Function;
+    width?: any;
```

Comment:
We can define this using style:{{width:"value"}}, do we explicitly need this? If width is passed from style, it will override this one.

Reply:
@psivakumaran  I agree with your point but the same column width will also get apply to corresponding cell. And currently I am not able to found any other way to get width of column from its style.

Reply:
done. now using width form style prop